hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaRolloutManagement.java

```diff
@@ -1147,4 +1152,26 @@ private void assertActionsPerTargetQuota(final Target target, final int requeste
         QuotaHelper.assertAssignmentQuota(target.getId(), requested, quota, Action.class, Target.class,
                 actionRepository::countByTargetId);
     }
+
+    private void runInUserContext(final BaseEntity rollout, final Runnable handler) {
```

Comment:
Due to isolated/generic nature of this method, I would extract it to either DeploymentHelper or even SystemManagement passing the userName parameter instead of BaseEntity. It will then help removing the duplication with AutoAssignChecker and potentially other classes

Reply:
Makes sense! I would design the generic helper method to accept a callback (basically a supplier) which can be invoked to obtain the "fallback user". Because - if the user context is already in place, there is no need to determine the fallback user.

Reply:
Moved to DeploymentHelper#runInNonSystemContext